### ISSUE TITLE
'categorical_feature in param dict is overridden.'

### ISSUE THREAD

**Issue by JYLFamily** [Microsoft/LightGBM#1408]
Why this warning？lgb.Dataset params is None. 
Thank you.

## Environment info
Operating System、CPU:Linux Server
C++/Python/R version:Python lightgbm 2.1.1

## Error Message:
warnings.warn('categorical_feature in param dict is overridden.')
## Reproducible examples
    
       def train(self, train_feature, train_label):
        train = self.__lgb.Dataset(
            data=train_feature,
            label=train_label,
            feature_name=train_feature.columns.tolist(),
            categorical_feature=self.__categorical_feature.tolist()
        )
        self.__clf = self.__lgb.train(
            self.__train_params,
            train,
            num_boost_round=self.__num_boost_round,
            categorical_feature=self.__categorical_feature.tolist()
        )

**Comment by StrikerRUS** [comment]
@JYLFamily This warnings is raised because you're passing categorical features twice: when constuct `Dataset` and when call `train`.

The correct way is to pass them only while constructing `Dataset` via `categorical_feature` argument.

**Comment by soerendip** [comment]
This warning appears also when the categorical features are only passed to train and not specified anywhere else.

**Comment by StrikerRUS** [comment]
> The correct way is to pass them only while constructing `Dataset` via `categorical_feature` argument.

```
import lightgbm as lgb

from sklearn.datasets import load_boston

X, y = load_boston(True)
lgb_data = lgb.Dataset(X, y, categorical_feature=[1, 2])
lgb.train({}, lgb_data)
```

```
D:\Miniconda3\lib\site-packages\lightgbm\basic.py:1247: UserWarning: Using categorical_feature in Dataset.
  warnings.warn('Using categorical_feature in Dataset.')
```

This warning is expected as per #792.

**Comment by guolinke** [comment]
@StrikerRUS 
maybe we can discard that warning when the data source is not pandas? I think the `auto` mode only works for pandas.

**Comment by StrikerRUS** [comment]
@guolinke 
> I think the auto mode only works for pandas.

> maybe we can discard that warning when the data source is not pandas?

OK, I'll take a look.

**Comment by elzizyel** [comment]
@StrikerRUS Warning still exsits.
<img width="1105" alt="pic" src="https://user-images.githubusercontent.com/42927184/65829376-9197c700-e2d7-11e9-90ed-b75857e7c98b.png">
